proposals/0304-structured-concurrency.md

```diff
@@ -763,30 +760,34 @@ extension Task {
 }
 ```
 
-The functions work the same as their instance counter parts, except that if invoked from a context that has no Task available, e.g. if invoked from outside of Swift's concurrency model (e.g. directly from a pthread) a default value is returned.
+These work the same as their instance counter parts, except that if invoked from a context that has no Task available, e.g. if invoked from outside of Swift's concurrency model (e.g. directly from a pthread) a default value is returned.
 
-The isCancelled function is implemented as:
+The `isCancelled` property is implemented as:
 
 ```swift
 extension Task {
   static var isCancelled: Bool { 
-    Task.current?.isCancelled ?? false
+    withUnsafeCurrentTask { task in
+      task?.isCancelled ?? false
+    }
   }
 }
 ```
 
 Which makes sense, because if not executing within a task, such code can never "be cancelled" using Swift's task infrastructure.
 
-This static `isCancelled` function is always safe to invoke, i.e. it may be invoked from synchronous or asynchronous functions and will always return the expected result. Do note however that checking cancellation while concurrently setting cancellation may be slightly racy, i.e. if the `cancel` is performed form another thread, the `isCancelled`
+This static `isCancelled` property is always safe to invoke, i.e. it may be invoked from synchronous or asynchronous functions and will always return the expected result. Do note however that checking cancellation while concurrently setting cancellation may be slightly racy, i.e. if the `cancel` is performed form another thread, the `isCancelled`
 
 #### Task priorities
 
-Similarly, a static `currentPriority` function is available to check the priority of the currently executing task:
+Similarly, a static `currentPriority` property is available to check the priority of the currently executing task:
 
 ```swift
 extension Task { 
   static var currentPriority: Task.Priority { 
-    Task.current?.priority ?? Task.Priority.default
```

Comment:
Please don't change this; the proposal is cleaner than the implementation and we don't need to show unsafe here.